File: AAKeyboardSchemaObjC/AAKeyboardSchemaObjC/ViewController.m
//
//  ViewController.m
//  AAKeyboardSchemaObjC
//
//  Created by sonson on 2015/02/09.
//  Copyright (c) 2015年 sonson. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
	[super viewDidLoad];
	// Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
	[super didReceiveMemoryWarning];
	// Dispose of any resources that can be recreated.
}

- (IBAction)registerAA:(id)sender {
	NSString *aa = @"　 　 　　　＿＿＿_ \n　 　　　／　　 　 　＼ \n　　　／　 _ノ 　ヽ､_　 ＼ \n　 ／ oﾟ(（●）) (（●）)ﾟo ＼ 　ほんとはVIPでやりたいんだお… \n　 |　　　　 （__人__）　　　　| \n　 ＼　　 　 ｀ ⌒´ 　 　 ／ \n";
	NSURLQueryItem *registerItem = [NSURLQueryItem queryItemWithName:@"register" value:aa];
	NSURLComponents *components = [[NSURLComponents alloc] initWithString:@"aakeyboard://app"];
	components.queryItems = @[registerItem];
	[[UIApplication sharedApplication] openURL:[components URL]];
}

- (IBAction)withCallback:(id)sender {
#if 1
	NSString *callback = @"aakeyboardsampleobjc://";
	NSString *aa = @"　 　 　　　＿＿＿_ \n　 　　　／　　 　 　＼ \n　　　／　 _ノ 　ヽ､_　 ＼ \n　 ／ oﾟ(（●）) (（●）)ﾟo ＼ 　ほんとはVIPでやりたいんだお… \n　 |　　　　 （__人__）　　　　| \n　 ＼　　 　 ｀ ⌒´ 　 　 ／ \n";
#else
	NSString *callback = @"hoge://";
	NSString *aa = @"＼(^o^)／";
#endif
	NSURLQueryItem *registerItem = [NSURLQueryItem queryItemWithName:@"register" value:aa];
	NSURLQueryItem *callbackItem = [NSURLQueryItem queryItemWithName:@"callback" value:callback];
	NSURLComponents *components = [[NSURLComponents alloc] initWithString:@"aakeyboard://app"];
	components.queryItems = @[registerItem, callbackItem];
	[[UIApplication sharedApplication] openURL:[components URL]];
}

@end

File: AAKeyboardSchemaSwift/AAKeyboardSchemaSwift/ViewController.swift
//
//  ViewController.swift
//  AAKeyboardSchemaSwift
//
//  Created by sonson on 2015/02/09.
//  Copyright (c) 2015年 sonson. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

	override func viewDidLoad() {
		super.viewDidLoad()
		// Do any additional setup after loading the view, typically from a nib.
	}

	override func didReceiveMemoryWarning() {
		super.didReceiveMemoryWarning()
		// Dispose of any resources that can be recreated.
	}

	@IBAction func registerAA(sender: AnyObject) {
		let text:String = "　 　 　　　＿＿＿_ \n　 　　　／　　 　 　＼ \n　　　／　 _ノ 　ヽ､_　 ＼ \n　 ／ oﾟ(（●）) (（●）)ﾟo ＼ 　ほんとはVIPでやりたいんだお… \n　 |　　　　 （__人__）　　　　| \n　 ＼　　 　 ｀ ⌒´ 　 　 ／ \n";
		let encoded:String = text.stringByAddingPercentEscapesUsingEncoding(NSUTF8StringEncoding)!;
		let URLString:String = "aakeyboard://app?register=" + encoded;
		UIApplication.sharedApplication().openURL(NSURL(string:URLString)!);
	}
	
	@IBAction func withCallback(sender: AnyObject) {
		let aa:String = "　 　 　　　＿＿＿_ \n　 　　　／　　 　 　＼ \n　　　／　 _ノ 　ヽ､_　 ＼ \n　 ／ oﾟ(（●）) (（●）)ﾟo ＼ 　ほんとはVIPでやりたいんだお… \n　 |　　　　 （__人__）　　　　| \n　 ＼　　 　 ｀ ⌒´ 　 　 ／ \n";
		let callback:String = "aakeyboardsampleswift://"
		let components:NSURLComponents = NSURLComponents(string: "aakeyboard://")!;
		let aaItem:NSURLQueryItem = NSURLQueryItem(name: "register", value: aa);
		let callbackItem:NSURLQueryItem = NSURLQueryItem(name: "callback", value: callback);
		components.queryItems = [aaItem, callbackItem];
		UIApplication.sharedApplication().openURL(components.URL!);
	}
	
	func hoge() {
		let aa:String = "＼(^o^)／";
		let callback:String = "hoge://"
		let components:NSURLComponents = NSURLComponents(string: "aakeyboard://")!;
		let aaItem:NSURLQueryItem = NSURLQueryItem(name: "register", value: aa);
		let callbackItem:NSURLQueryItem = NSURLQueryItem(name: "callback", value: callback);
		components.queryItems = [aaItem, callbackItem];
		UIApplication.sharedApplication().openURL(components.URL!);
		
	}
	
}

